src/Sentry/SpanTracer.cs

```diff
@@ -30,6 +31,16 @@ public class SpanTracer : ISpan
     /// <inheritdoc />
     public bool IsFinished => EndTimestamp is not null;
 
+    // Not readonly because of deserialization
+    private readonly Dictionary<string, Measurement> _measurements = new();
+
+    /// <inheritdoc />
+    public IReadOnlyDictionary<string, Measurement> Measurements => _measurements;
```

Comment:
I can do it easily enough in the `SpanTracer`. 

However, the `Span` and `Transaction` classes assign the measurements via a constructor... which makes the use of the Lazy class itself problematic. It would be possible to wire something up manually but not sure if this is worth doing since Span/Transaction are generally only ever created just before serializing the transaction and sending this over the wire. I guess it might save some allocations for events that get sampled out?